akka-actor-typed-tests/src/test/scala/akka/actor/typed/InterceptSpec.scala

```diff
@@ -28,16 +30,48 @@ object InterceptSpec {
       target(context, message)
     }
 
-    override def aroundSignal(
-        context: TypedActorContext[String],
-        signal: Signal,
-        target: SignalTarget[String]): Behavior[String] = {
-      target(context, signal)
-    }
-
     override def isSame(other: BehaviorInterceptor[Any, Any]): Boolean =
       other.isInstanceOf[SameTypeInterceptor]
   }
+
+  // This is similar to how EventSourcedBehavior is implemented
+  object MultiProtocol {
+    final case class ExternalResponse(s: String)
+    final case class Command(s: String)
+
+    sealed trait InternalProtocol
+    object InternalProtocol {
+      final case class WrappedCommand(c: Command) extends InternalProtocol
+      final case class WrappedExternalResponse(r: ExternalResponse) extends InternalProtocol
+    }
+
+    private class ProtocolTransformer extends BehaviorInterceptor[Any, InternalProtocol] {
+      override def aroundReceive(
+          ctx: TypedActorContext[Any],
+          msg: Any,
+          target: BehaviorInterceptor.ReceiveTarget[InternalProtocol]): Behavior[InternalProtocol] = {
+        val wrapped = msg match {
```

Comment:
this match would fail if another message comes e.g. a timer?

Reply:
The purpose of this is to test the same as we use in EventSourcedBehaviorImpl. There it's know that the incoming messages can either be ExternalProtocol or Command. Other unknown messages will blow up (but that would be a bug if other messages were sent to it). Doesn't matter if they blow up in the interceptor or in the receiveMessage of the inner Behavior.

Timer messages are not implemented with interceptors any more, they are unwrapped at the outer layer in ActorAdapter.